runtime/src/main/scala/akka/grpc/internal/ClientState.scala

```diff
@@ -22,138 +23,35 @@ import scala.util.Failure
 
 /**
  * INTERNAL API
- * Used from generated code so can't be private.
  *
  * Client utilities taking care of Channel reconnection and Channel lifecycle in general.
  */
 @InternalApi
-final class ClientState(
-    settings: GrpcClientSettings,
-    log: LoggingAdapter,
-    channelFactory: GrpcClientSettings => Future[InternalChannel])(implicit mat: Materializer, ex: ExecutionContext) {
-  def this(settings: GrpcClientSettings, log: LoggingAdapter)(implicit mat: Materializer, ex: ExecutionContext) =
-    this(settings, log, s => NettyClientUtils.createChannel(s, log))
-
-  // usually None, it'll have a value when the underlying InternalChannel is closing or closed.
-  private val closing = new AtomicReference[Option[Future[Done]]](None)
-  private val closeDemand: Promise[Done] = Promise[Done]()
+final class ClientState(settings: GrpcClientSettings, log: LoggingAdapter, channel: InternalChannel)(
+    implicit mat: Materializer,
+    ex: ExecutionContext) {
 
-  private val internalChannelRef = new AtomicReference[Option[Future[InternalChannel]]](Some(create()))
-  internalChannelRef.get().foreach(c => recreateOnFailure(c.flatMap(_.done), settings.creationAttempts))
+  @InternalStableApi
+  def this(settings: GrpcClientSettings, log: LoggingAdapter)(implicit mat: Materializer, ex: ExecutionContext) =
+    this(settings, log, NettyClientUtils.createChannel(settings, log))
 
   mat match {
     case m: ActorMaterializer =>
       m.system.whenTerminated.foreach(_ => close())(ex)
     case _ =>
   }
 
-  // used from generated client
+  // TODO #761 consider not leaking ManagedChannel to the generated code?
```

Comment:
This seems important. Is there a reason why users would need the "Managed" channel?

Reply:
I guess it was convenient at the time, but let's see if we can refactor that (in a different PR).